app/bundles/CoreBundle/Entity/CommonEntity.php

```diff
@@ -57,8 +65,8 @@ public function __toString()
     }
 
     /**
-     * @param string $prop
-     * @param mixed  $val
+     * @param string                                                                                                                                                                                                                                             $prop
+     * @param array<int|string, array<int|string, int|string>|int|string>|\DateTime|\DateTimeInterface|bool|int|string|self|FormEntity|IpAddress|TranslationEntityInterface|VariantEntityInterface|Category|Channel|FrequencyRule|Tag|Stage|User|Event|Role|null $val
```

Comment:
@shinde-rahul now that we defined all the types. I suggest we move the type of each entity into its respective entity

Reply:
I think that the `mixed` type was OK in this case. Imagine that this won't work for plugins for example

Reply:
@escopecz I figured out a solution https://github.com/mautic/mautic/pull/12497/commits/c37b3d21bcab8c158e6a761550599b6e7fc15b5a 

I was reading in `PHPStan` docs, they say in level 9 `mixed` can be only passed to `mixed` so the current format is not optimal